jarbas/core/management/commands/companies.py

```diff
@@ -38,52 +38,41 @@ def companies(self):
         Receives path to the dataset file and create a Company object for
         each row of each file. It creates the related activity when needed.
         """
-        skip=('main_acvtivity', 'secondary_activity')
+        skip=('main_activity','secondary_activity')
         keys = list(f.name for f in Company._meta.fields if f not in skip)
+        for num in range(1,100):
+            keys.append('secondary_activity_{}_code'.format(num))
+            keys.append('secondary_activity_{}'.format(num))
         with lzma.open(self.path, mode='rt') as file_handler:
             for row in csv.DictReader(file_handler):  
-               #main, secondary = self.return_activities(row)
-               
                filtered = {k: v for k, v in row.items() if k in keys}
-               
-               main = dict(
-                    code=row['main_activity_code'],
-                    description=row['main_activity']
+               data = dict(
+                       code=row['main_activity_code'],
+                       description=row['main_activity']
                )
-               
-               filtered['main_activity'] = [main]
-               
-               secondary = {}
-               for num in range(1,100):
-                   code = row.get('secondary_activity_{}_code'.format(num)) 
-                   description = row.get('secondary_activity_{}'.format(num))
-                   if code and description:
-                       secondary[code] = description
-
-               if len(secondary) == 0:
-                    filtered['secondary_activity'] = []
-               else:
-                    filtered['secondary_activity'] = [secondary]
-                 
+               filtered['main_activity'] = data
                obj = Company(**self.serialize(filtered))
                
                yield obj
                  
-    def return_activities(self, row):
-        data = dict(
-            code=row['main_activity_code'],
-            description=row['main_activity']
-        )
-
+    def serialize_activities(self, row):
         secondaries = {}
         for num in range(1, 100):
             code = row.get('secondary_activity_{}_code'.format(num))
             description = row.get('secondary_activity_{}'.format(num))
             if code and description:
                 secondaries[code] = description
-
-        return data, secondaries
-
+        
+        if len(secondaries) == 0:
+            row['secondary_activity'] = []
+        else:
+            row['secondary_activity'] = [secondaries]
+
+        for num in range(1,100):
```

Comment:
You can do this cleanup:

* with `del(row['secondary_activity_{}_code'.format(num)])` and `del(row['secondary_activity_{}'.format(num)])
* and within the loop of line 60 to avoid looping again ; )